lib/Sema/TypeCheckType.cpp

```diff
@@ -1212,6 +1212,56 @@ static Type diagnoseUnknownType(TypeResolution resolution,
   return ErrorType::get(ctx);
 }
 
+static bool shouldPreferModulesForTopLevelIdentTypeComponent(
+    const DeclContext *DC, TypeResolutionOptions options) {
+
+  // FIXME: We have a problem when a module and a type have the same name; the
+  // underlying UnqualifiedLookup code assumes that the type is always what's
+  // meant, even when that might not be the case. Work around this for
+  // parseable interfaces by making the *opposite* assumption outside of
+  // expressions and function bodies.
+  const SourceFile *containingSF = DC->getParentSourceFile();
```

Comment:
Nitpick: Can you make it more explicit here that this is just for parseable interfaces?